### PR TITLE
Gf1428 createopportunity

### PR DESCRIPTION
**Pull request by BenKDishman**
I think this is ready to go.  initial page from start page, Get form using query parameter (ok?).  
If we are moving to a manage controller class I'll refactor in the next story later today (1430).
Jon has stuck some lipstick on it. 
I think this is one review for merge but 3 relatively small questions come to mind though
1) are you happy for the get form to use query parameters ?
2) are you happy for the match to use 'hard coded' expected values for now ?
3) is the match in the createOpportunity function  okay or is there a more sophisticated way to do this ?

### CODE REVIEW

**Comment by jon-shipley** [comment]
I've also used the govuk selection-buttons.js code which depends on jQuery so that gets pulled in as well.   The selection buttons js code improves the styling  on the radio (and checkbox) buttons - they look a little ugly without it.

We will certainly want to do something with these growing js assets: e.g. concat, minify, gzip

File: src/main/twirl/views/wip.scala.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 @(backUrl:String)
 
 
-@main("Wip Page", backLink=Some(BackLink("Go Back", backUrl))) {
+@main("Wip Page", backLink=Some(BackLink("Go Back", backUrl)), displayUserName=Some("Experienced Eric")) {
```
Comment (kostya-lnk-ms) [inline src/main/twirl/views/wip.scala.html]: I'd rather define this as variable somewhere

File: src/main/resources/routes [hunk 2]
```diff
@@ -3,14 +3,16 @@
...
-GET         /opportunity/:opportunityId                  controllers.OpportunityController.showOpportunity(opportunityId:OpportunityId, section:Option[Int])
+GET         /                                                         controllers.StartPageController.startPage
+GET         /opportunities                                            controllers.OpportunityController.showOpportunities
+GET         /opportunities/guidance/:opportunityId                    controllers.OpportunityController.showGuidancePage(opportunityId:OpportunityId)
+GET         /opportunity/:opportunityId                               controllers.OpportunityController.showOpportunity(opportunityId:OpportunityId, section:Option[Int])
 
-GET         /application_form/:id/apply                  controllers.ApplicationController.showOrCreateForForm(id: ApplicationFormId)
+GET         /manage/opportunity/new                                   controllers.OpportunityController.createOpportunity
+
+GET         /application_form/:id/apply                               controllers.ApplicationController.showOrCreateForForm(id: ApplicationFormId)
```
Comment (kostya-lnk-ms) [inline src/main/resources/routes]: Should it be POST as this action modifies data?

File: src/main/resources/routes [hunk 3]
```diff
@@ -3,14 +3,16 @@
...
-GET         /opportunities/guidance/:opportunityId                      controllers.OpportunityController.showGuidancePage(opportunityId:OpportunityId)
-GET         /opportunity/:opportunityId                  controllers.OpportunityController.showOpportunity(opportunityId:OpportunityId, section:Option[Int])
+GET         /                                                         controllers.StartPageController.startPage
+GET         /opportunities                                            controllers.OpportunityController.showOpportunities
+GET         /opportunities/guidance/:opportunityId                    controllers.OpportunityController.showGuidancePage(opportunityId:OpportunityId)
+GET         /opportunity/:opportunityId                               controllers.OpportunityController.showOpportunity(opportunityId:OpportunityId, section:Option[Int])
 
-GET         /application_form/:id/apply                  controllers.ApplicationController.showOrCreateForForm(id: ApplicationFormId)
+GET         /manage/opportunity/new                                   controllers.OpportunityController.createOpportunity
```
Comment (kostya-lnk-ms) [inline src/main/resources/routes]: That should be PUT and also I'd rather have separate URLs for genuine "new"  & "re-use"

File: src/main/scala/controllers/OpportunityController.scala [hunk 4]
```diff
@@ -29,6 +30,17 @@ class OpportunityController @Inject()(opportunities: OpportunityOps, application
...
   }
 
+  def createOpportunity () = Action { implicit request =>
+    request.getQueryString("type").getOrElse("") match {
+      case "new" =>
+        Ok(views.html.wip(routes.OpportunityController.createOpportunity().url))
+      case "reuse" =>
+        Ok(views.html.wip(routes.OpportunityController.createOpportunity().url))
+      case "" =>
+        Ok(views.html.showNewOpportunityForm())
```
Comment (kostya-lnk-ms) [inline src/main/scala/controllers/OpportunityController.scala]: I'd rather avoid mixing "read-only" and actions, which actually modify some data.

**Comment by BenKDishman** [comment]
Hi Kostya, this has significantly changed on a code 'review' from Doug.  This is really just a navigation choice page which without GDS would probably be just a couple of links which is why it is using a GET form.  Hopefully function name changes should make this clearer.

The parameter for userName is rubbish I agree but this is really just a throwaway to get it looking as expected until we have the user logins integrated in a basic way.  Happy to change this in some way but I'm not quite clear how until we have a better idea of how this will operate ?

**Pull request merged**